src/clr-angular/forms/common/form.ts

```diff
@@ -28,4 +49,52 @@ export class ClrForm {
   markAsTouched() {
     this.markControlService.markAsTouched();
   }
+
+  @HostListener('submit')
+  onFormSubmit() {
+    this.markAsTouched();
+    // https://github.com/angular/angular/issues/12463
+    if (isPlatformBrowser(this.platformId)) {
+      const invalidControl = this.el.nativeElement.querySelector('.ng-invalid');
+      if (invalidControl) {
+        invalidControl.focus();
+        this.updateAriaLive();
+      } else {
+        this.resetAriaLive();
+      }
+    }
+  }
+
+  ngOnInit() {
+    // ClrForms need to have aria live element for screen readers.
+    this.ariaLiveElement = document.createElement('div');
+    this.ariaLiveElement.setAttribute('aria-live', 'polite');
+    this.ariaLiveElement.setAttribute('role', 'status');
+    this.ariaLiveElement.setAttribute('aria-atomic', 'true');
+    this.ariaLiveElement.setAttribute('aria-relevent', 'text');
+    this.ariaLiveElement.classList.add('clr-sr-only');
+    this.renderer.insertBefore(this.el.nativeElement, this.ariaLiveElement, this.el.nativeElement.firstChild);
+  }
+
+  private resetAriaLive() {
+    this.ariaLiveElement.innerHTML = '';
+  }
+
+  private updateAriaLive(): void {
+    let updatedLiveText = this.commonStrings.keys.formErrorSummary;
+    this.resetAriaLive();
+    const invalidControls = Array.prototype.slice.call(this.el.nativeElement.querySelectorAll('.ng-invalid'));
```

Comment:
The same issue here. This method is using native querying also relying on Angular's internal class. Also, I am really trying to understand what's going on here in this method. Is it trying to get all error messages and append them? Can we use:

```
@ContentChildren(ClrControlError, {descendants: true, read: ElementRef }) errors: QueryList<ElementRef>;

// then use Angular query
this.errors.map((erEl) => erEl.nativeElement.textContent) // this will give you all error messages in array.


```

Reply:
For every invalid control I need the projected content of the label (not the error message) so I can append it to the aria live region. The aria live region is supposed to read out a summary but not all of the details for each error.

Reply:
I left comments but they may only make sense to me. 
I'll try to explain again. 

1. I create an array of all invalid controls on the form (invalidControls)
2. then I make an array of only their id's (errorIDs)
3. Then I figure out which text to append from the values stored in the `this.labels`. 
4. Once I have an array or only the label text I can append it to a string that uses the formErrorSummary common string and update the aria live region. 

If forms switches to @ContentChildren this would change. But that depends on the other behavior that has to happen here which is putting focus on the first control that has an error state.